browser/components/translation/TranslationDocument.jsm

```diff
@@ -637,13 +643,35 @@ function swapTextForItem(item, target) {
           );
         }
 
-        // A trailing and a leading space must be preserved because
-        // they are meaningful in HTML.
-        let preSpace = /^\s/.test(curNode.nodeValue) ? " " : "";
-        let endSpace = /\s$/.test(curNode.nodeValue) ? " " : "";
-
-        curNode.nodeValue = preSpace + targetItem + endSpace;
-        curNode = getNextSiblingSkippingEmptyTextNodes(curNode);
+        // A workaround to show translated text with Quality Estimate annotations
+        // inplace (i.e. in the same webpage replacing the original text) and
+        // enabling seemless switch between showing original and translated text
+        // back and forth.
+        if (target == "original") {
+          // A trailing and a leading space must be preserved because
+          // they are meaningful in HTML.
+          let preSpace = /^\s/.test(curNode.nodeValue) ? " " : "";
+          let endSpace = /\s$/.test(curNode.nodeValue) ? " " : "";
+          curNode.nodeValue = preSpace + targetItem + endSpace;
+
+          for (let node of curNode.parentNode.childNodes) {
+            if (node.id == "QE-ANNOTATED") {
+              // There should be only 1 such node. Remove the curNode from the
```

Comment:
Could this fail and there be more than 1?

Reply:
Ideally it shouldn't because for any root TranslationItem (i.e. any sentence in source language with whatever no. of HTML tags in it), we generate only 1 Translation node. This statement holds true before this code change as well.
I tested this piece of code with console logs for few webpages (same ones that we used for demo yesterday) and I couldn't find more than 1 such nodes for any root TranslationItem.